cli/src/lib.rs

```diff
@@ -2100,6 +2103,31 @@ fn cluster_url(cfg: &Config) -> String {
     }
 }
 
+fn clean(cfg_override: &ConfigOverride) -> Result<()> {
+    let cfg = Config::discover(cfg_override)?.expect("Not in workspace.");
+    let cfg_parent = cfg.path().parent().expect("Invalid Anchor.toml");
+    let target_dir = cfg_parent.join("target");
+    let deploy_dir = target_dir.join("deploy");
+
+    for entry in fs::read_dir(target_dir)? {
+        let path = entry?.path();
+        if path.is_dir() && path != deploy_dir {
+            fs::remove_dir_all(path).expect("Could not remove directory.");
+        } else if path.is_file() {
+            fs::remove_file(path).expect("Could not remove file.");
+        }
+    }
+
+    for file in fs::read_dir(deploy_dir)? {
+        let path = file?.path();
+        if !path.to_string_lossy().ends_with("-keypair.json") {
```

Comment:
it's probably unlikely people will rename their json files here but to be safe, what do you think about changing this to just ".json"

Reply:
yeah, better to consider those cases as well. Maybe it's worth also checking that the .json is a valid keypair?

Reply:
dont think thats necessary. If someone put a json file there that is _not_ a valid keypair then thats probably intentional and we shouldnt delete that either